### PR TITLE
ipaddress: Use str.isascii() instead of frozenset.

### PR DESCRIPTION
**Pull request by methane**

### CODE REVIEW

File: Lib/ipaddress.py [hunk 1]
```diff
@@ -1764,16 +1763,22 @@ def _parse_hextet(cls, hextet_str):
               [0..FFFF].
 
         """
-        # Whitelist the characters, since int() allows a lot of bizarre stuff.
-        if not cls._HEX_DIGITS.issuperset(hextet_str):
-            raise ValueError("Only hex digits permitted in %r" % hextet_str)
+        # Reject non-ASCII characters, '+' and '-', since int() accepts them.
+        msg = "Only hex digits permitted in %r"
+        if not hextet_str.isascii() or not hextet_str.isalnum():
```
Comment (serhiy-storchaka) [inline Lib/ipaddress.py]: `'z'` and `'_'` shouldn't be permitted.

File: Lib/ipaddress.py [hunk 2]
```diff
@@ -488,8 +488,9 @@ def _prefix_from_prefix_string(cls, prefixlen_str):
         """
         # int allows a leading +/- as well as surrounding whitespace,
         # so we ensure that isn't the case
-        if not _BaseV4._DECIMAL_DIGITS.issuperset(prefixlen_str):
+        if not prefixlen_str.isascii() or not prefixlen_str.isdigit():
```
Comment (serhiy-storchaka) [inline Lib/ipaddress.py]: Is `_DECIMAL_DIGITS` still used?

**Comment by serhiy-storchaka** [comment]
What is the benefit of rewriting the code?

File: Lib/ipaddress.py (hunk 1 above)
Comment (methane) [reply Lib/ipaddress.py]: They are rejected by `int()`

Comment (serhiy-storchaka) [reply Lib/ipaddress.py]: `'0x12'` is accepted by `int()`.

Comment (methane) [reply Lib/ipaddress.py]: You're right.  I'll revert it.

**Comment by methane** [comment]
BTW, which is readable?

```
-        if not prefixlen_str.isascii() or not prefixlen_str.isdigit():
+        if not (prefixlen_str.isascii() and prefixlen_str.isdigit()):
```

**Comment by serhiy-storchaka** [comment]
I would prefer the second form. Both are compiled to the same bytecode.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**